### PR TITLE
Added a lucky number problem using C++ Language

### PR DESCRIPTION
**Pull request by hacker-4-good**

### CODE REVIEW

File: math/lucky_number.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        return true;
+    }
+}
+
+/**
+ * @brief Self-test implementations
+ * @return boolean
+*/
+
```
Comment (realstealthninja) [inline math/lucky_number.cpp:52]:
```suggestion
```

File: math/lucky_number.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    assert(math::isLuckyNumber(19) == true);
+    assert(math::isLuckyNumber(193) == true);
+    assert(math::isLuckyNumber(200) == false);
+    std::cout << "All test cases have successfully passed\n";
+}
+
+/**
+ * @brief User-defined test cases
+*/
+
```
Comment (realstealthninja) [inline math/lucky_number.cpp:64]:
```suggestion
```

**Review by realstealthninja — APPROVED** [review]
Wonderfully done! thank you for contributing

**Comment by hacker-4-good** [comment]
@realstealthninja when will 2nd person will approve it

**Comment by realstealthninja** [comment]
> @realstealthninja when will 2nd person will approve it

unsure. they are quite busy at the moment sorry for the long wait

**Pull request closed without merging**